python/cudf_polars/cudf_polars/dsl/ir.py

```diff
@@ -1653,11 +1653,36 @@ def do_evaluate(cls, schema: Schema, df: DataFrame) -> DataFrame:
 class MergeSorted(IR):
     """Merge sorted operation."""
 
-    def __init__(self, schema: Schema, left: IR, right: IR, key: str):
-        # libcudf merge is not stable wrt order of inputs, since
-        # it uses a priority queue to manage the tables it produces.
-        # See: https://github.com/rapidsai/cudf/issues/16010
-        raise NotImplementedError("MergeSorted not yet implemented")
+    __slots__ = ("key",)
+    _non_child = ("key",)
+    key: str
+    """Key that is sorted."""
+
+    def __init__(self, schema: Schema, key: str, left: IR, right: IR):
+        assert isinstance(left, Sort)
+        assert isinstance(right, Sort)
+        assert left.order == right.order
+        assert len(left.schema.keys()) <= len(right.schema.keys())
+        self.schema = schema
+        self.key = key
+        self.children = (left, right)
+        self._non_child_args = (key,)
```

Comment:
Is this defined above as `_non_child`? Can something be reused or simplified here?

Reply:
I think we need both. The `_non_child` arguments are needed to serialization/deserialization, among other things. While the `_non_child_args` are used to execute the IR nodes.